### PR TITLE
fix search URI, search param overwrite, and no results search render

### PR DESCRIPTION
**Pull request by steveinatorx**
## Description of changes

fix direct search URI load, render zero search results.

This still does a double tap call on basic List view (no query params). 

## Related issues (if any)

https://github.com/keystonejs/keystone/issues/3310

### CODE REVIEW

File: admin/client/App/screens/List/index.js [hunk 1]
```diff
@@ -63,10 +88,14 @@ const ListView = React.createClass({
...
+		// possibly specified query parameter
+		// parse query params first to get them into state
 		this.parseQueryParams();
-		this.loadItems();
+		this.initializeList(this.props.params.listId);
+		// suppress extraneous loadItems call when changing state via query params
+		if (Object.keys(this.props.location.query).length < 1) {
+			this.loadItems();
+		}
 	},
```
Comment (steveinatorx) [inline admin/client/App/screens/List/index.js]: heres the suppress

File: admin/client/App/screens/List/reducers/active.js [hunk 2]
```diff
@@ -29,11 +29,13 @@ const initialState = {
...
 		case SET_ACTIVE_LIST:
+
+			// prevent SET_ACTIVE_LIST from nuking the search state pulled in from the query string
 			return assign({}, state, {
 				id: action.id,
 				columns: action.list.expandColumns(action.list.defaultColumns),
 				filters: [],
-				search: '',
+				search: ((typeof state.id !== 'undefined') && (state.id !== action.id)) ? '' : state.search,
 				sort: action.list.expandSort(action.list.defaultSort),
```
Comment (steveinatorx) [inline admin/client/App/screens/List/reducers/active.js]: attempt to override store.js nuking the search state

File: admin/client/App/screens/Item/index.js [hunk 3]
```diff
@@ -75,7 +75,7 @@ var ItemView = React.createClass({
 	renderRelationships () {
 		const { relationships } = this.props.currentList;
 		const keys = Object.keys(relationships);
-		if (!keys.length) return;
+		if (!keys.length) return null;
 		return (
```
Comment (steveinatorx) [inline admin/client/App/screens/Item/index.js]: without the hard return null a load of a search query URI with zero results was getting a react render error

File: admin/client/App/screens/List/index.js (hunk 1 above)
Comment (steveinatorx) [inline admin/client/App/screens/List/index.js]: parseQuery first

**Comment by steveinatorx** [comment]
ok ive given up and just monkey patched my local fork with react-promises to get these double taps to at least return in the same order. From what i can tell the state machine in the latest admin app is pretty buggy. Someone in core team needs to refactor.

**Comment by mxstbr** [comment]
We're aware of that and working on it :wink: thanks for this PR, it's a pretty good reference!!

**Comment by steveinatorx** [comment]
cool. wish i had more time to help...

**Pull request closed without merging**